### PR TITLE
Build terraform image with populated provider plugins.

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

Created image to run terraform client. A terraform is called from a wrapper application tfcmt. Wrapper will post plan or apply commands results on GitHub PRs and commits. All plan command results published for older commits will be hidden by a github comment application. An image contains required plugins installed so they are not downloaded on each config run. This decrease startup time before each plan or apply commands execution.

On PR terraform plan will run, to let review command results.
On merge terraform apply will run, applying configs.

### CODE REVIEW

**Comment by dekiel** [comment]
Waiting for #6671 merge to run final end to end tests.

File: prow/jobs/test-infra/run-terraform-config.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      cluster: untrusted-workload
+      max_concurrency: 10
+      branches:
+        - ^master$
+        - ^main$
+      spec:
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/terraform:PR-6790"
+            securityContext:
+              privileged: true
```
Comment (Sawthis) [inline prow/jobs/test-infra/run-terraform-config.yaml:27]: Is privileged: true required in this case?

File: prow/jobs/test-infra/run-terraform-config.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
```
Comment (Sawthis) [inline prow/jobs/test-infra/run-terraform-config.yaml:74]: Is privileged: true required in this case?

File: prow/jobs/test-infra/run-terraform-config.yaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^master$
+        - ^main$
+      spec:
+        containers:
+          - image: "eu.gcr.io/kyma-project/test-infra/terraform:PR-6790"
+            securityContext:
+              privileged: true
```
Comment (Sawthis) [inline prow/jobs/test-infra/run-terraform-config.yaml:119]: Is privileged: true required in this case?

File: prow/jobs/test-infra/run-terraform-config.yaml (hunk 1 above)
Comment (dekiel) [reply prow/jobs/test-infra/run-terraform-config.yaml]: no

File: prow/jobs/test-infra/run-terraform-config.yaml (hunk 3 above)
Comment (dekiel) [reply prow/jobs/test-infra/run-terraform-config.yaml]: no

File: prow/jobs/test-infra/run-terraform-config.yaml (hunk 2 above)
Comment (dekiel) [reply prow/jobs/test-infra/run-terraform-config.yaml]: no

**Review by Sawthis — APPROVED** [review]

**Pull request merged**